### PR TITLE
Fixed a bug that caused arrays to be empty in some edge cases

### PR DESCRIPTION
**Pull request by earobinson**
```
templateFunction = jyson.buildTemplateFunction({
  a: [{
    a: 'a.$.a',
    b: 'b.b',
  }]
});
templateFunction({
  a: [{ a: 0 }, { a: 1 }],
  b: { b: false },
})
```
- would return `{ a: [] }` not `{ a: [ { a: 0, b: false }, { a: 1, b: false } ] }`
- added `.npmignore` file to keep package file smaller
- cleaned up some test cases

### CODE REVIEW

File: spec/lib/jyson/jyson.array.spec.js [hunk 1]
```diff
@@ -80,112 +80,110 @@ describe('jyson.array.spec: an array in the template', () => {
...
-          const json = this.templateFunction(input);
-          expect(json.a.length).to.equal(2);
-          expect(json.a[0].a).to.equal('a0a');
-          expect(json.a[0].b).to.equal(null);
-          expect(json.a[0].c).to.equal('a0c');
-          expect(json.a[1].a).to.equal('a1a');
-          expect(json.a[1].b).to.equal('a1b');
-          expect(json.a[1].c).to.equal('a1c');
-        });
+      it('must handel undefined values when converting an object to json', () => {
```
Comment (mshelley-hubba) [inline spec/lib/jyson/jyson.array.spec.js]: Minor: I think you meant to spell it ```handle```.

File: spec/lib/jyson/jyson.array.spec.js [hunk 2]
```diff
@@ -80,112 +80,110 @@ describe('jyson.array.spec: an array in the template', () => {
...
-          };
-          const json = this.templateFunction(input);
-          expect(json.a.length).to.equal(10);
-          for(let ii = 0; ii < 10; ii++) {
-            expect(json.a[ii].a).to.equal(`a${ii}a`);
-            expect(json.a[ii].b).to.equal(`a${ii}b`);
-            expect(json.a[ii].c).to.equal(`a${ii}c`);
-          }
-        });
+      it('must convert an object of length 10 to json', () => {
```
Comment (mshelley-hubba) [inline spec/lib/jyson/jyson.array.spec.js:130]: Why specifically length 10?

Comment (earobinson) [reply spec/lib/jyson/jyson.array.spec.js:130]: no reason its just a very large test, should it be named better?

Comment (mshelley-hubba) [reply spec/lib/jyson/jyson.array.spec.js:130]: I read it as a very specific thing that needed to happen. Right now it sounds like you *only* want objects of length 10 to convert to json.

Perhaps you just want to convert objects to json?

Comment (earobinson) [reply spec/lib/jyson/jyson.array.spec.js:130]: open to a better test name if you have an idea?

**Review by mshelley-hubba — APPROVED** [review]

**Pull request merged**